sdk/identity/azure-identity/test/e2e/azure_identity_e2e_test.cpp

```diff
@@ -3,32 +3,21 @@
 
 #include <azure/identity/managed_identity_credential.hpp>
 
+#include <azure/core/internal/environment.hpp>
+
 #include <chrono>
 #include <cstdlib>
 #include <exception>
 #include <iomanip>
 #include <iostream>
 #include <string>
 
-namespace {
-std::string GetEnv(std::string const& varName)
-{
-#if defined(_MSC_VER)
-#pragma warning(push)
-// warning C4996: 'getenv': This function or variable may be unsafe. Consider using _dupenv_s
-// instead.
-#pragma warning(disable : 4996)
-#endif
-  auto const value = std::getenv(varName.c_str());
-#if defined(_MSC_VER)
-#pragma warning(pop)
-#endif
-  return value ? std::string(value) : std::string();
-}
+using Azure::Core::_internal::Environment;
 
+namespace {
 std::string FormatEnvVarValue(std::string const& varName, bool isSecret)
 {
-  auto const value = GetEnv(varName);
+  auto const value = Environment::GetVariable(varName.c_str());
```

Comment:
What if we name it `AzureEnvironment` and implemented in a way it can ONLY read env vars which are prefixed as `AZURE_`  .

Makes me a little worry to tell customers that the SDK can be used to explore `env` with no restriction.

Reply:
Managed Identity Credential reads `MSI_ENDPOINT` variable.
Plus, I don't think this this removes the concern of us having to support it if publically exposed.